File: java/GeigerApiPluginConnector/src/main/java/eu/cybergeiger/api/plugin/PluginListener.java
package eu.cybergeiger.api.plugin;

import eu.cybergeiger.api.message.GeigerUrl;
import eu.cybergeiger.api.message.Message;

/**
 * <p>Interface for plugins listening for events.</p>
 */
public interface PluginListener {
  /**
   * <p>Listener for Geiger events.</p>
   *
   * <p>Any plugin must register for such events and </p>
   *
   * @param msg the message
   */
  void pluginEvent(Message msg);
}

File: dart/geiger_api/lib/src/plugin/plugin_listener.dart
library geiger_api;

import 'package:geiger_api/geiger_api.dart';

/// Interface for plugins listening for events.
abstract class PluginListener {
  /// Called when the listened event sends a [msg] to a [url].
  void pluginEvent(GeigerUrl? url, Message msg);
}
